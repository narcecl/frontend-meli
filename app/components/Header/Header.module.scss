.MeliHeader{
    background: #ffe600;
    padding: 10px 0;
}