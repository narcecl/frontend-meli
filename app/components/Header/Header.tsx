'use client';

import React, { useState } from 'react';
import Link from "next/link";
import { useSearchParams } from 'next/navigation';
import styles from './Header.module.scss';
import MercadoLibreLogo from '../../../assets/images/Logo_ML@2x.png';

import Search from '../Search/Search';

export const Header = () => {
    const searchParams = useSearchParams();
    const search = searchParams?.get('search') || '';
    const [ searchValue, setSearchValue ] = useState(search);

    return (
        <header className={styles.MeliHeader}>
            <div className="container">
                <div className="d-flex align-items-center gap-16">
                    <Link href="/">
                        <picture>
                            <img src={MercadoLibreLogo.src} alt="Logo MercadoLibre" width={48} />
                        </picture>
                    </Link>

                    <Search
                        value={searchValue}
                        onInput={setSearchValue}
                        placeholder="Nunca dejes de buscar"
                    />
                </div>
            </div>
        </header>  
    );
}

export default Header;