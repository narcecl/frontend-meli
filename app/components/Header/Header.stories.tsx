import type { Meta, StoryObj } from '@storybook/react';
import { Header } from './Header';

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
const meta: Meta<typeof Header> = {
    title: 'Core/Header',
    component: Header,
    tags: ['autodocs'],
    parameters: {
        // More on how to position stories at: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'fullscreen',
    },
    argTypes: {
    }
};

export default meta;
type Story = StoryObj<typeof Header>;

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default: Story = {
    args: {
    },
};