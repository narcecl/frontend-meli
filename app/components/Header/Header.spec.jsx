import { render } from '@testing-library/react';
import { useSearchParams } from 'next/navigation';
import '@testing-library/jest-dom';
import Header from './Header';

jest.mock('next/navigation', () => ({
	useSearchParams: jest.fn(),
}));

describe('Header', () => {
	it('renders the header component with search value from URL params', () => {
		// Mock the search value from URL params
		const mockSearchParams = new URLSearchParams('?search=query');
		useSearchParams.mockReturnValue(mockSearchParams);

		// Render the component
		const { container, asFragment } = render(<Header />);

		// Check first render
		const firstRender = asFragment();
        expect(firstRender).toMatchSnapshot();

		// Check if the logo is rendered
		const logoElement = container.querySelector('img');
		expect(logoElement).toBeInTheDocument();

		// Check if the search input is rendered with correct value
		const searchInput = container.querySelector('input');
		expect(searchInput).toBeInTheDocument();
		expect(searchInput).toHaveValue('query');
	});

	it('renders the header component with empty search value when URL params are not present', () => {
		// Mock empty search value from URL params
		useSearchParams.mockReturnValue(null);

		// Render the component
		const { container, asFragment } = render(<Header />);

		// Check first render
		const firstRender = asFragment();
        expect(firstRender).toMatchSnapshot();

		// Check if the logo is rendered
		const logoElement = container.querySelector('img');
		expect(logoElement).toBeInTheDocument();

		// Check if the search input is rendered with empty value
		const searchInput = container.querySelector('input');
		expect(searchInput).toBeInTheDocument();
		expect(searchInput).toHaveValue('');
	});
});
