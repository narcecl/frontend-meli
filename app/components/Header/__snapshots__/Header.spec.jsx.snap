// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the header component with empty search value when URL params are not present 1`] = `
<DocumentFragment>
  <header
    class="MeliHeader"
  >
    <div
      class="container"
    >
      <div
        class="d-flex align-items-center gap-16"
      >
        <a
          href="/"
        >
          <picture>
            <img
              alt="Logo MercadoLibre"
              width="48"
            />
          </picture>
        </a>
        <form
          action="/items"
          class="MeliSearch"
          method="GET"
        >
          <input
            class="MeliSearch__input"
            name="search"
            placeholder="Nunca dejes de buscar"
            value=""
          />
          <button
            aria-label="Buscar"
            class="MeliSearch__button"
            type="submit"
          />
        </form>
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Header renders the header component with search value from URL params 1`] = `
<DocumentFragment>
  <header
    class="MeliHeader"
  >
    <div
      class="container"
    >
      <div
        class="d-flex align-items-center gap-16"
      >
        <a
          href="/"
        >
          <picture>
            <img
              alt="Logo MercadoLibre"
              width="48"
            />
          </picture>
        </a>
        <form
          action="/items"
          class="MeliSearch"
          method="GET"
        >
          <input
            class="MeliSearch__input"
            name="search"
            placeholder="Nunca dejes de buscar"
            value="query"
          />
          <button
            aria-label="Buscar"
            class="MeliSearch__button"
            type="submit"
          />
        </form>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
